pkg/openstack/dataplane.go

```diff
@@ -62,7 +62,7 @@ func DataplaneNodesetsDeployed(version *corev1beta1.OpenStackVersion, dataplaneN
 		if !nodeset.IsReady() {
 			return false
 		}
-		if nodeset.Status.DeployedVersion != version.Spec.TargetVersion {
+		if !nodeset.Status.Conditions.IsTrue(dataplanev1.NodeSetMinorUpdateReadyCondition) {
```

Comment:
I think we might need to check both of these right?

Reply:
The version check has already been done in the nodeset controller before the condition is set.

Reply:
while that may be true, this function gets executed from the openstackversion side of things which could bump targetVersion. I am trying to guard against the case where the version changes, but in the interim this condition hasn't yet been reset

Reply:
Reworked so that this "if" condition will work without any changes.